bash_completionsV2.go

```diff
@@ -220,7 +220,14 @@ __%[1]s_handle_standard_completion_case() {
 
     # Short circuit to optimize if we don't have descriptions
     if [[ "${completions[*]}" != *$tab* ]]; then
-        IFS=$'\n' read -ra COMPREPLY -d '' < <(compgen -W "${completions[*]}" -- "$cur")
+        compgen_words=$(printf "%%s\n" "${completions[@]}")
+        IFS=$'\n' read -ra COMPREPLY -d '' < <(IFS=$'\n' compgen -W "${compgen_words}" -- "$cur")
```

Comment:
Could this cause the performance loss?

This also feels bit clumsy to me, but I was not able to get completions with whitespace to be separate indices of `COMPREPLY` without `printf` here.

Reply:
It's almost certainly here. One starting point for debugging would be to just with strategically placed `time`s, e.g. `time compgen_words=...`.

`printf -v "arr[index]" ...` in a loop would avoid a subshell, but the net result could be even slower as this doesn't create subshells in a loop.

Reply:
I don't think we need `printf` at all. Wouldn't something like this work?

```bash
IFS=$'\n' read -ra COMPREPLY -d '' < <(IFS=$'\n'; compgen -W "${completions[*]}" -- "$cur")
```
`IFS=$'\n' compgen ...` sets `IFS` for just the `compgen` call.

`IFS=$'\n'; compgen ...` sets `IFS` for both the `compgen` call and the `"${completions[*]}"` string

And since that change is in a subshell, it won't affect `IFS` for the rest of the script

----

You can see what this does with this simpler example:

```bash
$ arr=( 1 2 3 )
$ IFS=$'\n' echo "${arr[*]}"
$ IFS=$'\n'; echo "${arr[*]}"
```